components/public-api-server/main.go

```diff
@@ -21,11 +22,17 @@ func main() {
 		logger.WithError(err).Fatal("Failed to initialize public api server.")
 	}
 
+	if listenErr := srv.ListenAndServe(); listenErr != nil {
+		logger.WithError(listenErr).Fatal("Failed to serve public api server")
+	}
+}
+
+func register(srv *baseserver.Server) error {
 	srv.HTTPMux().HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
nit: `r` is unused here; can be replaced with `_`.

Reply:
I need to get around to configuring the linter to do this for us :)

Reply:
Fixed.